### PR TITLE
Sort SQL migration steps with a stable ordering

### PR DESCRIPTION
**Pull request by tomhoule**
closes: https://github.com/prisma/prisma-engines/issues/1942

### CODE REVIEW

File: migration-engine/connectors/sql-migration-connector/src/sql_schema_differ.rs [hunk 1]
```diff
@@ -61,48 +60,39 @@ pub(crate) fn calculate_steps(schemas: Pair<&SqlSchema>, flavour: &dyn SqlFlavou
...
+                index: idxs.as_ref().map(|(_, idx)| *idx),
+            }))
+            .chain(
+                redefine_indexes
+                    .into_iter()
+                    .map(|idxs| SqlMigrationStep::RedefineIndex {
+                        table: idxs.as_ref().map(|(table, _)| *table),
+                        index: idxs.as_ref().map(|(_, idx)| *idx),
+                    }),
+            ),
```
Comment (tomhoule) [inline migration-engine/connectors/sql-migration-connector/src/sql_schema_differ.rs:92]: The code is left unchanged there, but we can start simplifying after this PR, now that we don't need to go overboard with iterators anymore.

File: migration-engine/connectors/sql-migration-connector/src/sql_migration.rs [hunk 2]
```diff
@@ -28,23 +28,37 @@ impl DatabaseMigrationMarker for SqlMigration {
     }
 }
 
-#[derive(Debug)]
+// The order of the variants matters for sorting.
```
Comment (pimeys) [inline migration-engine/connectors/sql-migration-connector/src/sql_migration.rs:31]: how?

Comment (tomhoule) [reply migration-engine/connectors/sql-migration-connector/src/sql_migration.rs:31]: `derive(Ord)` will sort the steps by the variant first. I'll add that to the comment

Comment (pimeys) [reply migration-engine/connectors/sql-migration-connector/src/sql_migration.rs:31]: so, wait, enum ordering sorts by the name of the enum? what about the data?

Comment (tomhoule) [reply migration-engine/connectors/sql-migration-connector/src/sql_migration.rs:31]: Not by name but order of declaration.

Data is sorted by integer index. It's stable and it's the same order as in the source, so it's good for us.

**Review by pimeys — APPROVED** [review]

**Pull request merged**